packages/astro-parser/src/parse/utils/features.ts

```diff
@@ -0,0 +1,2 @@
+
+export const FEATURE_CUSTOM_ELEMENT = 1 << 0;
```

Comment:
No, this allows you to pack multiple boolean values into a single integer. It's equivalent to `{ hasCustomElements: true, hasOtherFeature: true }`. I just felt that using bits was appropriate here since this is the parser and bitwise is basically free, no cost added when parsing the majority of astro files that don't have custom elements.

Reply:
Nice! I have no problem with this approach, I just have never wrapped my head around the general concept.

Reply:
Ok cool, I find them useful some times in low-level use-cases like this. I think Preact is using them now. Next time we talk I'll teach you about them, as this part of it is simple to understand.